File: app/src/main/java/net/bible/android/view/activity/readingplan/DailyReadingItemAdapter.java
/*
 * Copyright (c) 2018 Martin Denham, Tuomas Airaksinen and the And Bible contributors.
 *
 * This file is part of And Bible (http://github.com/AndBible/and-bible).
 *
 * And Bible is free software: you can redistribute it and/or modify it under the
 * terms of the GNU General Public License as published by the Free Software Foundation,
 * either version 3 of the License, or (at your option) any later version.
 *
 * And Bible is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
 * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 * See the GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License along with And Bible.
 * If not, see http://www.gnu.org/licenses/.
 *
 */

package net.bible.android.view.activity.readingplan;

import java.util.List;

import net.bible.service.readingplan.OneDaysReadingsDto;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TwoLineListItem;

/**
 * Retain similar style to TwoLineListView but for single TextView on each line
 * 
 * @author Martin Denham [mjdenham at gmail dot com]
 */
public class DailyReadingItemAdapter extends ArrayAdapter<OneDaysReadingsDto> {

	private int resource;

	public DailyReadingItemAdapter(Context _context, int _resource, List<OneDaysReadingsDto> _items) {
		super(_context, _resource, _items);
		resource = _resource;
	}

	@Override
	public View getView(int position, View convertView, ViewGroup parent) {

		OneDaysReadingsDto item = getItem(position);

		// Pick up the TwoLineListItem defined in the xml file
		TwoLineListItem view;
		if (convertView == null) {
			LayoutInflater inflater = (LayoutInflater) getContext().getSystemService(Context.LAYOUT_INFLATER_SERVICE);
			view = (TwoLineListItem) inflater.inflate(resource, parent, false);
		} else {
			view = (TwoLineListItem) convertView;
		}

		// Set value for the first text field
		if (view.getText1() != null) {
			String line1 = item.getDayDesc();
			view.getText1().setText(line1);
		}

		// set value for the second text field
		if (view.getText2() != null) {
			String line2 = item.getReadingsDesc();
			view.getText2().setText(line2);
		}
		
		return view;
	}
}

File: app/src/main/java/net/bible/android/view/activity/readingplan/DailyReadingItemAdapter.kt
/*
 * Copyright (c) 2020-2022 Martin Denham, Tuomas Airaksinen and the AndBible contributors.
 *
 * This file is part of AndBible: Bible Study (http://github.com/AndBible/and-bible).
 *
 * AndBible is free software: you can redistribute it and/or modify it under the
 * terms of the GNU General Public License as published by the Free Software Foundation,
 * either version 3 of the License, or (at your option) any later version.
 *
 * AndBible is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
 * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 * See the GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License along with AndBible.
 * If not, see http://www.gnu.org/licenses/.
 */

package net.bible.android.view.activity.readingplan

import net.bible.service.readingplan.OneDaysReadingsDto
import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TwoLineListItem
import net.bible.android.activity.databinding.TwoLineListItemBinding
import java.text.SimpleDateFormat

/**
 * @author Martin Denham [mjdenham at gmail dot com]
 */
class DailyReadingItemAdapter(_context: Context, private val resource: Int, _items: List<OneDaysReadingsDto>) :
		ArrayAdapter<OneDaysReadingsDto>(_context, resource, _items) {

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {

        val item = getItem(position)

        val binding = if (convertView == null)
            TwoLineListItemBinding.inflate(context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater, parent, false)
        else TwoLineListItemBinding.bind(convertView)
        val view = convertView ?: binding.root
        item ?: return view

        // Set value for the first text field
        binding.text1.text = if (item.isDateBasedPlan && item.readingDate != null) {
            SimpleDateFormat.getDateInstance().format(item.readingDate!!)
        } else {
            item.dayDesc
        }

        // set value for the second text field
        val line2 = item.readingsDesc
        binding.text2.text = line2

        return view
    }
}
